src/compiler/transformers/ts.ts

```diff
@@ -877,6 +903,8 @@ namespace ts {
                 // the body of a class with static initializers.
                 setEmitFlags(classExpression, EmitFlags.Indented | getEmitFlags(classExpression));
                 expressions.push(startOnNewLine(createAssignment(temp, classExpression)));
+                addRange(expressions, pendingExpressions);
+                pendingExpressions = savedPendingExpressions;

```

Comment:
You save and reset unconditionally, you should restore them unconditionally as well and not only in the true branch of an `if`.